compiler/optimizer/SwitchAnalyzer.cpp

```diff
@@ -1262,8 +1262,13 @@ TR::Block *TR::SwitchAnalyzer::addTableBlock(SwitchInfo *dense)
    int32_t branchTable = 0;
 
    TR::Node *node = TR::Node::create(_switch, TR::table, 3 + upperBound + branchTable);
+   // X and P use safeToSkipTableBounds to remove the bounds check
+   // Z reads safeToSkipTableBounds but appears to use cannotOverflow to remove the bounds check
```

Comment:
@fjeremic fyi for this difference on Z
@knn-k and @0xdaryl for the arm codegen behaviour

Reply:
I've opened #3769 to address the Z portion.